frontend/src/app/forms/side-menu/side-menu.component.ts

```diff
@@ -1,43 +1,61 @@
-/**
- * @description
- *  This component renders the hamburger side menu and its actions.
- */
+/*
+  @description
+    This component renders the hamburger side menu and its actions.
+*/
 
-import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Component, Input } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { FormGroup } from '../../common/forms/forms';
+import { FormsData } from '../../model/forms-data.model';
+import { ApplicationStep, ApplicationStepsArr } from '../model/steps.model';
 
 @Component({
   selector: 'app-side-menu',
   templateUrl: './side-menu.component.html',
   styleUrls: ['./side-menu.component.scss'],
 })
 export class SideMenuComponent {
+  constructor(private activatedRoute: ActivatedRoute, private router: Router) {}
+
+  /**
+   * Make the Steps available in the template.
+   *
+   * Only access this property in the template.
+   * In the TS file, you can directly refer to the underlying element.
+   */
+  readonly APPLICATION_STEP = ApplicationStep;
+
+  /**
+   * Make the Steps Array available in the template.
+   *
+   * Only access this property in the template.
+   * In the TS file, you can directly refer to the underlying element.
+   */
+  readonly APPLICATION_STEP_ARR = ApplicationStepsArr;
+
   /**
-   * This array holds the list of different steps information.
+   * Data of the entire form.
    */
-  @Input() sideMenuList: Array<{
-    id: number;
-    displayName: string;
-    isCompleted: boolean;
-    isActive: boolean;
-    selector: string;
-  }>;
+  @Input()
+  formsData: FormGroup<FormsData>;
 
   /**
-   * This takes care of emitting event to parent.
+   * Which step is currently displayed?
    */
-  @Output() private chosenStep = new EventEmitter();
+  @Input()
+  currentStep: ApplicationStep;
 
   /**
-   * This method emits the event for step navigation.
-   * @param  menu - Contains the menu to navigate
+   * Update the "step" query paremeter.
+   * You can use this to navigate between the different form steps!
    */
-  changeStep(menu: {
-    id: number;
-    displayName: string;
-    isCompleted: boolean;
-    isActive: boolean;
-    selector: string;
-  }): void {
-    this.chosenStep.emit(menu);
+  navigateToStep(step: ApplicationStep): void {
```

Comment:
I prefer not to repeat the code and handle step transition in one place by emitting event to parent method.

Reply:
I don't really have a strong opinion about that. To me, it feels like the purpose of the router is, among others, to allow changing the route in various places. But we can also emit an event to the parent. :)